docs/source/community/pull_requests.rst

```diff
@@ -0,0 +1,360 @@
+#. Comment on the code when issues are spotted.
+
+   - Is the code using existing infrastructure to the extent possible?
+   - Is the code accruing technical debt?
+   - Is the formatting consistent? (Don't worry about formatting which CI will
+     fix)
+
+#. When the PR is marked as ready to go, complete a final pass through the code
+   flagging any potential issues.
+#. If issues arise work with the author to resolve them. Repeating the previous
+   step as necessary.
+#. If you are the last reviewer to approve a PR then merge it (assuming all
+   CI workflows have passed).
+
+Notes on PR Quality
+===================
+
+.. note::
+
+   The contents of this section are written assuming a 1.0 has occurred. We
+   admittedly have not lived up to the lofty standards of this section and
+   part of getting to a 1.0 is making sure existing code meets or exceeds
+   these standards.
+
+.. note::
+
+   Occasional contributors from outside the project are not the target of this
+   section. This section is targeting developers who are part of the team.
+
+The checklist on the PR template is admittedly asking a lot of the author.
+NWChemEx is designed to be a modular code. The vast majority of electronic
+structure development occurs in modules. Each of these modules are disjoint,
+and can be separately hosted. In turn, you can (and should) go through the
+"just get something working" phase outside of the NWChemEx repositories.
+
+By making a PR which will add a module into an NWX repository you are saying the
+module is ready for users. This does not necessarily mean that it is as
+performant as it is going to get, or that the module is fully featured. It
+simply means that the contents of the module have been vetted for correctness
+and the module has been found potentially useful. Once the module is merged it
+becomes available to users, and since taking it back would break any code that
+uses the module, we as a project are obligated to support that module or break
+API. The point being, we expect the module to be fully documented, tested, and
+to adhere to the organization's standards so that we can begin stewardship of
+the module.
+
+For PRs addressing infrastructure the requirements are a bit different.
+Since infrastructure exists outside modules, it can't really be staged until
+it is ready. For infrastructure we will allow some technical debt, as long as
+the developer is willing to repay that technical debt, and makes a plan for
+repaying it. In practice what this means is, say you really need a new type of
+cache to complete a run. We'll let you merge that cache, say without
+documentation, as long as you open an issue tracking that documentation is
+still needed. Developers are expected to repay technical debt on a relatively
+timely basis. That said, since the infrastructure is going into the main
+repository, it needs to still be vetted. This means it can not still be in the
+design phase, nor can it be untested.
```

Comment:
So discussions about the infrastructure should be primarily in issues?

Reply:
Maybe? PR discussions should focus on the code in the PR and how to change that code in order to  merge it; theoretically these are things like "you didn't document this" or "change this to X to get CI to pass". Issues are meant for hashing out all of the details needed to write a PR, so "what algorithm should I use" or "what should this property type look like". 

In reality this isn't as clear cut as it sounds because if you don't realize that something in your PR will be a problem, then you end up debating the merits of say an algorithm or design in the PR, whereas our delimitation would suggest it should happen in an issue. I think the solution is that some "issue" discussion in a PR is okay, but if it goes on too long it needs to be moved to an issue and tackled there.